lang/java/reef-common/src/main/java/org/apache/reef/client/DriverLauncher.java

```diff
@@ -59,15 +60,21 @@
       .build();
 
   private final REEF reef;
+  private final UserCredentials user;
 
   private LauncherStatus status = LauncherStatus.INIT;
 
   private String jobId;
   private RunningJob theJob;
 
   @Inject
-  private DriverLauncher(final REEF reef) {
+  private DriverLauncher(final REEF reef, final UserCredentials user) {
```

Comment:
`DriverLauncher` *is* the client. The credentials are set e.g. here: [YarnSubmissionHelper:296](https://github.com/apache/reef/pull/1290/files#diff-0561fcf2f2b2fa1fae4a3037ccd79a2bR296)
That happens somewhere inside the `DriverLauncher.submit()` call; after that, a call to `DriverLauncher.getUser()` will return a proxy user endowed with YARN credentials to be used on the driver side.